code/mouse.py

```diff
@@ -148,20 +148,25 @@ def mouse_trigger_zoom_mouse():
         if eye_zoom_mouse.zoom_mouse.enabled:
             eye_zoom_mouse.zoom_mouse.on_pop(eye_zoom_mouse.zoom_mouse.state)
 
-    def mouse_drag():
+    def mouse_drag(button: int):
         """(TEMPORARY) Press and hold/release button 0 depending on state for dragging"""
         # todo: fixme temporary fix for drag command
         button_down = len(list(ctrl.mouse_buttons_down())) > 0
         # print(str(ctrl.mouse_buttons_down()))
         if not button_down:
             # print("start drag...")
-            ctrl.mouse_click(button=0, down=True)
+            ctrl.mouse_click(button=button, down=True)
             # app.notify("drag started")
         else:
+            # End any active drags
             # print("end drag...")
-            ctrl.mouse_click(button=0, up=True)
+            self.end_drag(0)
```

Comment:
At this point, you can safely look at ctrl.mouse_buttons_down() to end the correct drag(s). The relevant fix to make the button indices consistent across platforms was released in v0.2.

It might make sense to make end_drag parameterless?